### ISSUE TITLE
Fixture with customisation creates empty object

### ISSUE THREAD

**Issue by jk-idealo** [Nylle/JavaFixture#75]
Take the following class
```java
    public class MyTestObject {
        private String myString;
        private Integer myInteger;
        private List<String> myList = new ArrayList<>();
        private Map<Integer, Integer> myMap = new HashMap<>();

        public MyTestObject() {
        }

        public String getMyString() {
            return myString;
        }

        public void setMyString(String myString) {
            this.myString = myString;
        }

        public Integer getMyInteger() {
            return myInteger;
        }

        public void setMyInteger(Integer myInteger) {
            this.myInteger = myInteger;
        }

        public List<String> getMyList() {
            return myList;
        }

        public void setMyList(List<String> myList) {
            this.myList = myList;
        }

        public Map<Integer, Integer> getMyMap() {
            return myMap;
        }

        public void setMyMap(Map<Integer, Integer> myMap) {
            this.myMap = myMap;
        }
    }
```
If you create a fixture without customisation, it works, so the following test is **green**:
```java
    @Test
    void create() {
        var result = fixture().create(MyTestObject.class);

        assertThat(result.getMyInteger()).isNotNull();
        assertThat(result.getMyString()).isNotNull();
        assertThat(result.getMyList()).isNotEmpty();
        assertThat(result.getMyMap()).isNotEmpty();
    }
```
However, when customising `myInteger`, the following 4 assertions are all **red**:
```java
    @Test
    void customise() {
        var result = fixture().build(MyTestObject.class)
                .with("myInteger", 0)
                .create();

        assertThat(result.getMyInteger()).isEqualTo(0);
        assertThat(result.getMyString()).isNotNull();
        assertThat(result.getMyList()).isNotEmpty();
        assertThat(result.getMyMap()).isNotEmpty();
    }
```
It _looks_ like the no-args-constructor is being used right away.
(It worked in version 2.8.x.)

**Comment by akutschera** [comment]
That's an interesting one.
I found a simpler test to reproduce it:
```java
void jf75() {
            var result = fixture().build(TestObjectWithGenerics.class)
                    .with("optional", Optional.empty())
                    .create();

            assertThat(result.getOptional()).isEmpty();
        }
```
This is actually caused by the fix to [#63](https://github.com/Nylle/JavaFixture/issues/63).
We currently don't have a mechanism to specify which object of a nested tree has a property called "optional" (which fails object creation due to [#49](https://github.com/Nylle/JavaFixture/issues/49))
Therefore the ObjectSpecimen-creation fails because one of the objects in the tree does not have this property.
Because of the fallback we now use any available public constructor which is the empty constructor in most cases.

**Comment by iljakorneckis** [comment]
Another example, without Optional and where the property exists:

```java
@Test
void xxx() {
    Fixture sut = new Fixture(configuration);
    var expected = new TestObjectWithGenerics();
    var result = sut.build(TestObjectWithDeepNesting.class)
            .with("nested", expected)
            .create();
    var actual = result.getNested();
    assertThat(actual).isNotNull();
}
```
